### ISSUE TITLE
libjnidispatch.so linking issues - support for Graal VM native testing 

### ISSUE THREAD

**Issue by cforce** [testcontainers/testcontainers-java#6297]
### Module

None

### Problem

I am developing a spring-cloud-function (spring boot) based  aws lambda  app which is compiled  with Graal image and  shall be tested as native  compiled  application as well using maven plugin.
https://graalvm.github.io/native-build-tools/latest/maven-plugin.html

While running native testing the  below issue appears which i assume is based on missing reachability information for testcontainers in graalvm reachability metadata repository https://github.com/oracle/graalvm-reachability-metadata/tree/master/metadata/org.testcontainers/testcontainers

 org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy$LibC.<clinit> -> libjnidispatch.so

```
[2/7] Performing analysis...  [**********]                                                             (189.4s @ 4.34GB)
  15,363 (89.01%) of 17,260 classes reachable
  25,425 (65.46%) of 38,839 fields reachable
  69,945 (57.00%) of 122,717 methods reachable
     493 classes,   153 fields, and 2,506 methods registered for reflection
      65 classes,    68 fields, and    56 methods registered for JNI access
       4 native libraries: dl, pthread, rt, z
```

```
8:47:22.229 [main] DEBUG org.testcontainers.dockerclient.DockerClientProviderStrategy - UnixSocketClientProviderStrategy: failed with exception RuntimeException (org.testcontainers.shaded.com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.github.dockerjava.api.model.Info
: no suitable constructor found, can not deserialize from Object value (missing default constructor or creator, or perhaps need to add/enable type information?)
 at [Source: N/A; line: -1, column: -1]). Root cause JsonMappingException (Can not construct instance of com.github.dockerjava.api.model.Info: no suitable constructor found, can not deserialize from Object value (missing default constructor or creator, or perhaps need to add/enable type information?)       
 at [Source: N/A; line: -1, column: -1])
08:47:22.229 [main] DEBUG org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy - $XDG_RUNTIME_DIR is not set.
08:47:22.229 [main] DEBUG org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy - '/root/.docker/run/docker.sock' does not exist.

Failures (1):
  JUnit Jupiter:GetSessionsTest
    ClassSource [className = 'demo.cloudfunctiondynamodblambda.GetSessionsTest', filePosition = null]
    => java.lang.UnsatisfiedLinkError: Native library (com/sun/jna/linux-x86-64/libjnidispatch.so) not found in resource path ()
       com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1033)
       com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:989)
       com.sun.jna.Native.<clinit>(Native.java:195)
       org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy$LibC.<clinit>(RootlessDockerClientProviderStrategy.java:96)
       org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy.lambda$null$0(RootlessDockerClientProviderStrategy.java:37)
       [...]
       Suppressed: java.lang.NoClassDefFoundError: Could not initialize class org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy$LibC
         org.testcontainers.dockerclient.RootlessDockerClientProviderStrategy.lambda$null$0(RootlessDockerClientProviderStrategy.java:37)
[         0 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         0 tests failed          ]

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  09:52 min
[INFO] Finished at: 2022-12-07T08:47:22+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.19:test (test-native) on project cloud-function-dynamodb-lambda: Execution of /git/serverless/spring-native-aws-demo/cloud-function-dynamodb-lambda/target/native-tests --xml-output-dir /git/serverless/spring-native-aws-demo/cloud-
function-dynamodb-lambda/target/native-test-reports -Djunit.platform.listeners.uid.tracking.output.dir=/git/serverless/spring-native-aws-demo/cloud-function-dynamodb-lambda/target/test-ids returned non-zero result -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

### Solution

Create tests to cover the graal native image based compilation and testing  to detect issues and regression test for supporting thee spring boot way of native testing 
Add necesarry [

2/7] Performing analysis...  [**********]                                                             (189.4s @ 4.34GB)
  15,363 (89.01%) of 17,260 classes reachable
  25,425 (65.46%) of 38,839 fields reachable
  69,945 (57.00%) of 122,717 methods reachable
     493 classes,   153 fields, and 2,506 methods registered for reflection
      65 classes,    68 fields, and    56 methods registered for JNI access
       4 native libraries: dl, pthread, rt, z

### Benefit

Graal won't go away  ;) Testing features platform specific is necessary to allow discover platform specific tests

### Alternatives

-

### Would you like to help contributing this feature?

yes. Added possible fix as below fork, but trying to test that now.

**Comment by cforce** [comment]
Sure, but i tried several jetpack configs but actually none o them worked for me
Something i must be doin wrong
```
		<dependency>
			<groupId>com.github.mhalbritter</groupId>
			<artifactId>graalvm-reachability-metadata</artifactId>
			<version>mh/testcontainers</version>
		</dependency>
```

```
		<dependency>
			<groupId>com.github.oracle</groupId>
			<artifactId>graalvm-reachability-metadata</artifactId>
			<version>PR132</version>
		</dependency>
```
```
		<dependency>
			<groupId>com.github.oracle</groupId>
			<artifactId>graalvm-reachability-metadata</artifactId>
			<version>PR132-SNAPSHOT</version>
		</dependency>
```

Its still a mystery for me what needs to be used  :)
Besides that the PR is accepted but not yet mergd into master, is it . At least i don't find that extra line in the resources cfg on master.

**Comment by eddumelendez** [comment]
I gave it a try yesterday. The `graalvm-reachability-metadata` should execute `./gradlew package` and then in your pom file

```xml
<profiles>
    <profile>
        <id>native</id>
        <build>
            <pluginManagement>
                <plugins>
                    <plugin>
                        <groupId>org.springframework.boot</groupId>
                        <artifactId>spring-boot-maven-plugin</artifactId>
                        <configuration>
                            <image>
                                <builder>paketobuildpacks/builder:tiny</builder>
                                <env>
                                    <BP_NATIVE_IMAGE>true</BP_NATIVE_IMAGE>
                                </env>
                            </image>
                        </configuration>
                        <executions>
                            <execution>
                                <id>process-aot</id>
                                <goals>
                                    <goal>process-aot</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
                    <plugin>
                        <groupId>org.graalvm.buildtools</groupId>
                        <artifactId>native-maven-plugin</artifactId>
                        <configuration>
                            <classesDirectory>${project.build.outputDirectory}</classesDirectory>
                            <metadataRepository>
                                <enabled>true</enabled>
                            </metadataRepository>
                            <requiredVersion>22.3</requiredVersion>
                        </configuration>
                        <executions>
                            <execution>
                                <id>add-reachability-metadata</id>
                                <goals>
                                    <goal>add-reachability-metadata</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
                </plugins>
            </pluginManagement>
        </build>
    </profile>
    <profile>
        <id>nativeTest</id>
        <dependencies>
            <dependency>
                <groupId>org.junit.platform</groupId>
                <artifactId>junit-platform-launcher</artifactId>
                <scope>test</scope>
            </dependency>
        </dependencies>
        <build>
            <plugins>
                <plugin>
                    <groupId>org.springframework.boot</groupId>
                    <artifactId>spring-boot-maven-plugin</artifactId>
                    <executions>
                        <execution>
                            <id>process-test-aot</id>
                            <goals>
                                <goal>process-test-aot</goal>
                            </goals>
                        </execution>
                    </executions>
                </plugin>
                <plugin>
                    <groupId>org.graalvm.buildtools</groupId>
                    <artifactId>native-maven-plugin</artifactId>
                    <configuration>
                        <classesDirectory>${project.build.outputDirectory}</classesDirectory>
                        <metadataRepository>
                            <enabled>true</enabled>
                            <url>file:/Users/username//graalvm-reachability-metadata/build/graalvm-reachability-metadata-0.2.6-SNAPSHOT.zip</url>
                        </metadataRepository>
                        <requiredVersion>22.3</requiredVersion>
                    </configuration>
                    <executions>
                        <execution>
                            <id>native-test</id>
                            <goals>
                                <goal>test</goal>
                            </goals>
                        </execution>
                    </executions>
                </plugin>
            </plugins>
        </build>
    </profile>
</profiles>
```

Note the `url` tag at `metadataRepository`.

Just sharing my findings so far

* tc database driver works
* tests with junit jupiter don’t work
```
Caused by: java.lang.IllegalArgumentException: Could not find class [com.example.springbootpostgresqlflyway.PostgisTest__ApplicationContextInitializer]
       org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:333)
       org.springframework.context.aot.AotApplicationContextInitializer.instantiateInitializer(AotApplicationContextInitializer.java:80)
       org.springframework.context.aot.AotApplicationContextInitializer.initialize(AotApplicationContextInitializer.java:71)
       org.springframework.context.aot.AotApplicationContextInitializer.lambda$forInitializerClasses$0(AotApplicationContextInitializer.java:61)
       org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:603)
       [...]
     Caused by: java.lang.ClassNotFoundException: com.example.springbootpostgresqlflyway.PostgisTest__ApplicationContextInitializer
       java.base@17.0.5/java.lang.Class.forName(DynamicHub.java:1132)
       org.springframework.util.ClassUtils.forName(ClassUtils.java:283)
       org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:323)
```

* test with access host such an example prometheus or selenium don’t work
```
Caused by: java.io.IOException: Key exchange was not finished, connection is closed.
        at org.testcontainers.shaded.com.trilead.ssh2.transport.KexManager.getOrWaitForConnectionInfo(KexManager.java:93)
        at org.testcontainers.shaded.com.trilead.ssh2.transport.TransportManager.getConnectionInfo(TransportManager.java:231)
        at org.testcontainers.shaded.com.trilead.ssh2.Connection.connect(Connection.java:711)
        ... 60 common frames omitted
Caused by: java.io.IOException: Fatal error during MAC startup!
        at org.testcontainers.shaded.com.trilead.ssh2.transport.KexManager.finishKex(KexManager.java:299)
        at org.testcontainers.shaded.com.trilead.ssh2.transport.KexManager.handleMessage(KexManager.java:623)
        at org.testcontainers.shaded.com.trilead.ssh2.transport.TransportManager.receiveLoop(TransportManager.java:756)
        at org.testcontainers.shaded.com.trilead.ssh2.transport.TransportManager$1.run(TransportManager.java:470)
        at java.base@17.0.5/java.lang.Thread.run(Thread.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
```

**Comment by cforce** [comment]
Tx for the profile!

It seems like the nativTest profile with a local build reachability zip based on 

commit d6128d04f819cb1a8e35a338d6d8db63a7aaa0d5 (HEAD -> master, origin/master, origin/HEAD)
Merge: 7053700 e4177ed
Author: Fabio Niephaus <fabio.niephaus@oracle.com>
Date:   Wed Dec 7 17:24:24 2022 +0100


mvn package -PnativeTestX
....
....
....
```
.testcontainers.lang":"java","org.testcontainers.sessionId":"b96d62c3-369b-4387-9b16-575cfb58a0f0","org.testcontainers.version":"1.17.6"}},"NetworkSettings":{"Bridge":"","SandboxID":"cdacec0d6848e7666f9f5486929e0e09d168273ebea62e595c5c4b31cc210aed","HairpinMode":false,"LinkLocalIPv6Address":"","LinkLocalIPv
6PrefixLen":0,"Ports":{},"SandboxKey":"/var/run/docker/netns/cdacec0d6848","SecondaryIPAddresses":null,"SecondaryIPv6Addresses":null,"EndpointID":"","Gateway":"","GlobalIPv6Address":"","GlobalIPv6PrefixLen":0,"IPAddress":"","IPPrefixLen":0,"IPv6Gateway":"","MacAddress":"","Networks":{"bridge":{"IPAMConfig":
null,"Links":null,"Aliases":null,"NetworkID":"7210c323b8a95144ea2cfb53ebde0eca61763eca3856369ec93a33a9d3f821c1","EndpointID":"","Gateway":"","IPAddress":"","IPPrefixLen":0,"IPv6Gateway":"","GlobalIPv6Address":"","GlobalIPv6PrefixLen":0,"MacAddress":"","DriverOpts":null}}}}[\n]"
10:58:39.781 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "[\r][\n]"
10:58:39.781 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "0[\r][\n]"
10:58:39.781 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ep-00000011: releasing valid endpoint
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000011: releasing endpoint
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000011: connection http-outgoing-1 can be kept alive for 3 MINUTES
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000011: connection released [route: {}->unix://localhost:2375][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
10:58:39.782 [main] DEBUG org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: f5218a121281a62f36957309d6a13a77f852b5bd3fc19d1b49325f76d651b735,true,true
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ex-00000013: preparing request execution
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.protocol.RequestAddCookies - Cookie spec selected: strict
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.protocol.RequestAuthCache - Auth cache not set in the context
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ProtocolExec - ex-00000013: target auth state: UNCHALLENGED
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ProtocolExec - ex-00000013: proxy auth state: UNCHALLENGED
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ConnectExec - ex-00000013: acquiring connection with route {}->unix://localhost:2375
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ex-00000013: acquiring endpoint (3 MINUTES)
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ex-00000013: endpoint lease request (3 MINUTES) [route: {}->unix://localhost:2375][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]   
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ex-00000013: endpoint leased [route: {}->unix://localhost:2375][total available: 0; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ex-00000013: acquired ep-00000012
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ex-00000013: acquired endpoint ep-00000012
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.MainClientExec - ex-00000013: executing DELETE /v1.32/containers/f5218a121281a62f36957309d6a13a77f852b5bd3fc19d1b49325f76d651b735?v=true&force=true HTTP/1.1
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ep-00000012: start execution ex-00000013
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000012: executing exchange ex-00000013 over http-outgoing-1
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> DELETE /v1.32/containers/f5218a121281a62f36957309d6a13a77f852b5bd3fc19d1b49325f76d651b735?v=true&force=true HTTP/1.1
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> accept: application/json
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> x-tc-sid: b96d62c3-369b-4387-9b16-575cfb58a0f0
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> Accept-Encoding: gzip, x-gzip, deflate
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> Host: localhost:2375
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> Connection: keep-alive
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 >> User-Agent: Apache-HttpClient/5.0.3 (Java/17.0.5)
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "DELETE /v1.32/containers/f5218a121281a62f36957309d6a13a77f852b5bd3fc19d1b49325f76d651b735?v=true&force=true HTTP/1.1[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "accept: application/json[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "x-tc-sid: b96d62c3-369b-4387-9b16-575cfb58a0f0[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "Accept-Encoding: gzip, x-gzip, deflate[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "Host: localhost:2375[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "Connection: keep-alive[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "User-Agent: Apache-HttpClient/5.0.3 (Java/17.0.5)[\r][\n]"
10:58:39.782 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 >> "[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "HTTP/1.1 204 No Content[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "Api-Version: 1.41[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "Date: Fri, 09 Dec 2022 09:58:39 GMT[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "Docker-Experimental: false[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "Ostype: linux[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "Server: Docker/20.10.21 (linux)[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.wire - http-outgoing-1 << "[\r][\n]"
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << HTTP/1.1 204 No Content
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << Api-Version: 1.41
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << Date: Fri, 09 Dec 2022 09:58:39 GMT
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << Docker-Experimental: false
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << Ostype: linux
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.headers - http-outgoing-1 << Server: Docker/20.10.21 (linux)
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.MainClientExec - ex-00000013: connection can be kept alive for 3 MINUTES
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient - ep-00000012: releasing valid endpoint
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000012: releasing endpoint
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000012: connection http-outgoing-1 can be kept alive for 3 MINUTES
10:58:39.788 [main] DEBUG com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager - ep-00000012: connection released [route: {}->unix://localhost:2375][total available: 1; route allocated: 1 of 2147483647; total allocated: 1 of 2147483647]
10:58:39.788 [main] DEBUG org.testcontainers.utility.ResourceReaper - Removed container and associated volume(s): amazon/dynamodb-local:latest
com.mercedes.mic.demo.cloudfunctiondynamodblambda.GetSessionsTest > [1] userId=player19 SUCCESSFUL

com.mercedes.mic.demo.cloudfunctiondynamodblambda.GetSessionsTest > [2] userId=player42 SUCCESSFUL

com.mercedes.mic.demo.cloudfunctiondynamodblambda.GetSessionsTest > [1] userId=player1 SUCCESSFUL

com.mercedes.mic.demo.cloudfunctiondynamodblambda.GetSessionsTest > [2] userId=player3 SUCCESSFUL


Test run finished after 148627 ms
[         4 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         4 tests successful      ]
[         0 tests failed          ]

[INFO]
[INFO] --- maven-jar-plugin:3.3.0:jar (default-jar) @ cloud-function-dynamodb-lambda ---
[INFO] 
[INFO] --- spring-boot-maven-plugin:3.0.0:repackage (repackage) @ cloud-function-dynamodb-lambda ---
^[[6~^[[6~^[[6~[INFO] Attaching repackaged archive /git/serverless/spring-native-aws-demo/cloud-function-dynamodb-lambda/target/cloud-function-dynamodb-lambda-0.0.1-SNAPSHOT-exec.jar with classifier exec
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  19:56 min
[INFO] Finished at: 2022-12-09T10:58:45+01:00
[INFO] ------------------------------------------------------------------------

```

I wonder when i use your "mvn package -Pnative" i can't  even trigger the graal native image build - i don't see any logs of it as i do when use nativeTest - why is that?

**Comment by eddumelendez** [comment]
Run `mvn -Pnative native:compile`. Please check the [docs](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html#native-image.developing-your-first-application.native-build-tools.maven)
